packages/runtime/src/templates/getHandler.ts

```diff
@@ -67,9 +73,7 @@ const makeHandler = (conf: NextConfig, app, pageRoot, staticManifest: Array<[str
     }
     const url = new URL(event.rawUrl)
     const port = Number.parseInt(url.port) || 80
-    const { host } = event.headers
-    const protocol = event.headers['x-forwarded-proto'] || 'http'
-    base = `${protocol}://${host}`
+    base = url.origin
```

Comment:
[`origin`](https://developer.mozilla.org/en-US/docs/Web/API/URL/origin) is `protocol://host:port`